In [1]:
#importaciones de las librerias
import pandas as pd
from sklearn.metrics import confusion_matrix
from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from IPython.core import display as disp

In [2]:
data=pd.read_csv("cancer_mama.csv")
data

clase     edad menopausia tamaño capsulas  \
no-recurrence-events  30-39  premeno      30-34    0-2       no   
no-recurrence-events  40-49  premeno      20-24    0-2       no   
no-recurrence-events  40-49  premeno      20-24    0-2       no   
no-recurrence-events  60-69     ge40      15-19    0-2       no   
no-recurrence-events  40-49  premeno        0-4    0-2       no   
...                     ...      ...        ...    ...      ...   
recurrence-events     30-39  premeno      30-34    0-2       no   
recurrence-events     30-39  premeno      20-24    0-2       no   
recurrence-events     60-69     ge40      20-24    0-2       no   
recurrence-events     40-49     ge40      30-34    3-5       no   
recurrence-events     50-59     ge40      30-34    3-5       no   

                      grado_maligno  pecho zona_pecho radioterapia  
no-recurrence-events              3   left   left_low           no  
no-recurrence-events              2  right   right_up           no  
no-recurrence-events              2   left   left_low           no  
no-recurrence-events              2  right    left_up           no  
no-recurrence-events              2  right  right_low           no  
...                             ...    ...        ...          ...  
recurrence-events                 2   left    left_up           no  
recurrence-events                 3   left    left_up          yes  
recurrence-events                 1  right    left_up           no  
recurrence-events                 3   left   left_low           no  
recurrence-events                 3   left   left_low           no  

[286 rows x 9 columns]

In [3]:
data.columns

Index(['clase', 'edad', 'menopausia', 'tamaño', 'capsulas', 'grado_maligno',
       'pecho', 'zona_pecho', 'radioterapia'],
      dtype='object')

# 1. Realiza todo el preprocesamiento necesario para poder entrenar el clasificador y muestra las distintas tablas de distribución.

### Preprocesamiento de los datos

In [4]:
data.info()#no hay missing values

<class 'pandas.core.frame.DataFrame'>
Index: 286 entries, no-recurrence-events to recurrence-events
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   clase          286 non-null    object
 1   edad           286 non-null    object
 2   menopausia     286 non-null    object
 3   tamaño         286 non-null    object
 4   capsulas       286 non-null    object
 5   grado_maligno  286 non-null    int64 
 6   pecho          286 non-null    object
 7   zona_pecho     286 non-null    object
 8   radioterapia   286 non-null    object
dtypes: int64(1), object(8)
memory usage: 22.3+ KB


In [5]:
data = pd.get_dummies(data) #de variables categoricas a numericas en distintas columnas
data

grado_maligno  clase_20-29  clase_30-39  clase_40-49  \
no-recurrence-events              3            0            1            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            1   
...                             ...          ...          ...          ...   
recurrence-events                 2            0            1            0   
recurrence-events                 3            0            1            0   
recurrence-events                 1            0            0            0   
recurrence-events                 3            0            0            1   
recurrence-events                 3            0            0            0   

                      clase_50-59  clase_60-69  clase_70-79  edad_ge40  \
no-recurrence-events            0            0            0          0   
no-recurrence-events            0            0            0          0   
no-recurrence-events            0            0            0          0   
no-recurrence-events            0            1            0          1   
no-recurrence-events            0            0            0          0   
...                           ...          ...          ...        ...   
recurrence-events               0            0            0          0   
recurrence-events               0            0            0          0   
recurrence-events               0            1            0          1   
recurrence-events               0            0            0          1   
recurrence-events               1            0            0          1   

                      edad_lt40  edad_premeno  ...  pecho_left  pecho_right  \
no-recurrence-events          0             1  ...           1            0   
no-recurrence-events          0             1  ...           0            1   
no-recurrence-events          0             1  ...           1            0   
no-recurrence-events          0             0  ...           0            1   
no-recurrence-events          0             1  ...           0            1   
...                         ...           ...  ...         ...          ...   
recurrence-events             0             1  ...           1            0   
recurrence-events             0             1  ...           1            0   
recurrence-events             0             0  ...           0            1   
recurrence-events             0             0  ...           1            0   
recurrence-events             0             0  ...           1            0   

                      zona_pecho_?  zona_pecho_central  zona_pecho_left_low  \
no-recurrence-events             0                   0                    1   
no-recurrence-events             0                   0                    0   
no-recurrence-events             0                   0                    1   
no-recurrence-events             0                   0                    0   
no-recurrence-events             0                   0                    0   
...                            ...                 ...                  ...   
recurrence-events                0                   0                    0   
recurrence-events                0                   0                    0   
recurrence-events                0                   0                    0   
recurrence-events                0                   0                    1   
recurrence-events                0                   0                    1   

                      zona_pecho_left_up  zona_pecho_right_low  \
no-recurrence-events                   0                     0   
no-recurrence-events                   0                     0   
no-recurrence-events                   0                     0   
no-recurrence-events              

In [6]:
data.columns#Mostramos todas las columnas 

Index(['grado_maligno', 'clase_20-29', 'clase_30-39', 'clase_40-49',
       'clase_50-59', 'clase_60-69', 'clase_70-79', 'edad_ge40', 'edad_lt40',
       'edad_premeno', 'menopausia_0-4', 'menopausia_10-14',
       'menopausia_15-19', 'menopausia_20-24', 'menopausia_25-29',
       'menopausia_30-34', 'menopausia_35-39', 'menopausia_40-44',
       'menopausia_45-49', 'menopausia_5-9', 'menopausia_50-54', 'tamaño_0-2',
       'tamaño_12-14', 'tamaño_15-17', 'tamaño_24-26', 'tamaño_3-5',
       'tamaño_6-8', 'tamaño_9-11', 'capsulas_?', 'capsulas_no',
       'capsulas_yes', 'pecho_left', 'pecho_right', 'zona_pecho_?',
       'zona_pecho_central', 'zona_pecho_left_low', 'zona_pecho_left_up',
       'zona_pecho_right_low', 'zona_pecho_right_up', 'radioterapia_no',
       'radioterapia_yes'],
      dtype='object')

In [7]:
print(data['zona_pecho_?'])#Se han creado columnas sin informacion relevante

no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
                       ..
recurrence-events       0
recurrence-events       0
recurrence-events       0
recurrence-events       0
recurrence-events       0
Name: zona_pecho_?, Length: 286, dtype: uint8


In [8]:
#Mostramos todas las rows de esta columna para demostrar que es irrelevante
pd.set_option('display.max_rows', data.shape[0]+1)
print(data['capsulas_?'])

no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrence-events    0
no-recurrenc

In [9]:
data=data.drop(["capsulas_?",'zona_pecho_?'],axis=1)#Eliminamos las columnas irrelevantes

In [10]:
#dado que lo que se han creado dos columnas para radioterapia, una radioterapia_yes y otra radioterapia_no
# y dado que una es la contraria de la otra optaremos por mantener radioterapia_yes
data=data.drop(["radioterapia_no"],axis=1)

In [11]:
data #Mostramos los datos tras el preprocesamiento

grado_maligno  clase_20-29  clase_30-39  clase_40-49  \
no-recurrence-events              3            0            1            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            1   
no-recurrence-events              3            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              3            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              3            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            1            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            1            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              1            0            0            1   
no-recurrence-events              1            0            0            1   
no-recurrence-events              2         

### Tablas distribucion

In [12]:
#for que recorre todas las columnas del dataframe de los datos
#Muestra las tablas de frecuencias con la columna radioterapia_yes con el resto de columnas
#Utilizamos la funcion display para mostrar varios dataframes
#Para mostrar las tablas de distribucion hacemos uso de la funcion de pandas.crosstab
for x in data.columns:
    disp.display(pd.crosstab(index=data[x], columns=data['radioterapia_yes']))

radioterapia_yes   0   1
grado_maligno           
1                 64   7
2                 98  32
3                 56  29

radioterapia_yes    0   1
clase_20-29              
0                 217  68
1                   1   0

radioterapia_yes    0   1
clase_30-39              
0                 192  58
1                  26  10

radioterapia_yes    0   1
clase_40-49              
0                 154  42
1                  64  26

radioterapia_yes    0   1
clase_50-59              
0                 137  53
1                  81  15

radioterapia_yes    0   1
clase_60-69              
0                 177  52
1                  41  16

radioterapia_yes    0   1
clase_70-79              
0                 213  67
1                   5   1

radioterapia_yes    0   1
edad_ge40                
0                 118  39
1                 100  29

radioterapia_yes    0   1
edad_lt40                
0                 211  68
1                   7   0

radioterapia_yes    0   1
edad_premeno             
0                 107  29
1                 111  39

radioterapia_yes    0   1
menopausia_0-4           
0                 210  68
1                   8   0

radioterapia_yes    0   1
menopausia_10-14         
0                 193  65
1                  25   3

radioterapia_yes    0   1
menopausia_15-19         
0                 194  62
1                  24   6

radioterapia_yes    0   1
menopausia_20-24         
0                 177  59
1                  41   9

radioterapia_yes    0   1
menopausia_25-29         
0                 181  51
1                  37  17

radioterapia_yes    0   1
menopausia_30-34         
0                 174  52
1                  44  16

radioterapia_yes    0   1
menopausia_35-39         
0                 203  64
1                  15   4

radioterapia_yes    0   1
menopausia_40-44         
0                 203  61
1                  15   7

radioterapia_yes    0   1
menopausia_45-49         
0                 217  66
1                   1   2

radioterapia_yes    0   1
menopausia_5-9           
0                 215  67
1                   3   1

radioterapia_yes    0   1
menopausia_50-54         
0                 213  65
1                   5   3

radioterapia_yes    0   1
tamaño_0-2               
0                  35  38
1                 183  30

radioterapia_yes    0   1
tamaño_12-14             
0                 218  65
1                   0   3

radioterapia_yes    0   1
tamaño_15-17             
0                 213  67
1                   5   1

radioterapia_yes    0   1
tamaño_24-26             
0                 218  67
1                   0   1

radioterapia_yes    0   1
tamaño_3-5               
0                 199  51
1                  19  17

radioterapia_yes    0   1
tamaño_6-8               
0                 210  59
1                   8   9

radioterapia_yes    0   1
tamaño_9-11              
0                 215  61
1                   3   7

radioterapia_yes    0   1
capsulas_no              
0                  30  34
1                 188  34

radioterapia_yes    0   1
capsulas_yes             
0                 190  40
1                  28  28

radioterapia_yes    0   1
pecho_left               
0                 101  33
1                 117  35

radioterapia_yes    0   1
pecho_right              
0                 117  35
1                 101  33

radioterapia_yes      0   1
zona_pecho_central         
0                   199  66
1                    19   2

radioterapia_yes       0   1
zona_pecho_left_low         
0                    136  40
1                     82  28

radioterapia_yes      0   1
zona_pecho_left_up         
0                   146  43
1                    72  25

radioterapia_yes        0   1
zona_pecho_right_low         
0                     201  61
1                      17   7

radioterapia_yes       0   1
zona_pecho_right_up         
0                    191  62
1                     27   6

radioterapia_yes    0   1
radioterapia_yes         
0                 218   0
1                   0  68

In [13]:
#Tablas de distribucion normalizadas por columnas, es decir, divide el count values que podemos ver en la
#anterior tabla entre la suma de los count values de cada columna.
for x in data.columns:
    disp.display(pd.crosstab(index=data[x], columns=data['radioterapia_yes'], normalize='columns'))

radioterapia_yes         0         1
grado_maligno                       
1                 0.293578  0.102941
2                 0.449541  0.470588
3                 0.256881  0.426471

radioterapia_yes         0    1
clase_20-29                    
0                 0.995413  1.0
1                 0.004587  0.0

radioterapia_yes         0         1
clase_30-39                         
0                 0.880734  0.852941
1                 0.119266  0.147059

radioterapia_yes         0         1
clase_40-49                         
0                 0.706422  0.617647
1                 0.293578  0.382353

radioterapia_yes        0         1
clase_50-59                        
0                 0.62844  0.779412
1                 0.37156  0.220588

radioterapia_yes         0         1
clase_60-69                         
0                 0.811927  0.764706
1                 0.188073  0.235294

radioterapia_yes         0         1
clase_70-79                         
0                 0.977064  0.985294
1                 0.022936  0.014706

radioterapia_yes         0         1
edad_ge40                           
0                 0.541284  0.573529
1                 0.458716  0.426471

radioterapia_yes        0    1
edad_lt40                     
0                 0.96789  1.0
1                 0.03211  0.0

radioterapia_yes         0         1
edad_premeno                        
0                 0.490826  0.426471
1                 0.509174  0.573529

radioterapia_yes         0    1
menopausia_0-4                 
0                 0.963303  1.0
1                 0.036697  0.0

radioterapia_yes         0         1
menopausia_10-14                    
0                 0.885321  0.955882
1                 0.114679  0.044118

radioterapia_yes         0         1
menopausia_15-19                    
0                 0.889908  0.911765
1                 0.110092  0.088235

radioterapia_yes         0         1
menopausia_20-24                    
0                 0.811927  0.867647
1                 0.188073  0.132353

radioterapia_yes         0     1
menopausia_25-29                
0                 0.830275  0.75
1                 0.169725  0.25

radioterapia_yes         0         1
menopausia_30-34                    
0                 0.798165  0.764706
1                 0.201835  0.235294

radioterapia_yes         0         1
menopausia_35-39                    
0                 0.931193  0.941176
1                 0.068807  0.058824

radioterapia_yes         0         1
menopausia_40-44                    
0                 0.931193  0.897059
1                 0.068807  0.102941

radioterapia_yes         0         1
menopausia_45-49                    
0                 0.995413  0.970588
1                 0.004587  0.029412

radioterapia_yes         0         1
menopausia_5-9                      
0                 0.986239  0.985294
1                 0.013761  0.014706

radioterapia_yes         0         1
menopausia_50-54                    
0                 0.977064  0.955882
1                 0.022936  0.044118

radioterapia_yes        0         1
tamaño_0-2                         
0                 0.16055  0.558824
1                 0.83945  0.441176

radioterapia_yes    0         1
tamaño_12-14                   
0                 1.0  0.955882
1                 0.0  0.044118

radioterapia_yes         0         1
tamaño_15-17                        
0                 0.977064  0.985294
1                 0.022936  0.014706

radioterapia_yes    0         1
tamaño_24-26                   
0                 1.0  0.985294
1                 0.0  0.014706

radioterapia_yes         0     1
tamaño_3-5                      
0                 0.912844  0.75
1                 0.087156  0.25

radioterapia_yes         0         1
tamaño_6-8                          
0                 0.963303  0.867647
1                 0.036697  0.132353

radioterapia_yes         0         1
tamaño_9-11                         
0                 0.986239  0.897059
1                 0.013761  0.102941

radioterapia_yes         0    1
capsulas_no                    
0                 0.137615  0.5
1                 0.862385  0.5

radioterapia_yes        0         1
capsulas_yes                       
0                 0.87156  0.588235
1                 0.12844  0.411765

radioterapia_yes         0         1
pecho_left                          
0                 0.463303  0.485294
1                 0.536697  0.514706

radioterapia_yes         0         1
pecho_right                         
0                 0.536697  0.514706
1                 0.463303  0.485294

radioterapia_yes           0         1
zona_pecho_central                    
0                   0.912844  0.970588
1                   0.087156  0.029412

radioterapia_yes            0         1
zona_pecho_left_low                    
0                    0.623853  0.588235
1                    0.376147  0.411765

radioterapia_yes           0         1
zona_pecho_left_up                    
0                   0.669725  0.632353
1                   0.330275  0.367647

radioterapia_yes             0         1
zona_pecho_right_low                    
0                     0.922018  0.897059
1                     0.077982  0.102941

radioterapia_yes            0         1
zona_pecho_right_up                    
0                    0.876147  0.911765
1                    0.123853  0.088235

radioterapia_yes    0    1
radioterapia_yes          
0                 1.0  0.0
1                 0.0  1.0

In [14]:
#Eliminamos 5 rows para el ultimo apartado
test = data.tail(5)
data.drop(data.tail(5).index,inplace=True)

# 2. Crea un clasificador e indica su error. Úsalo para saber a qué clase corresponden al menos 5 pacientes que no hayas usado para entrenar el modelo

He decidido utilizar bayes como clasificador dado que en uno de los apartados se nos pide sacar las tablas de distribucion y por ello he entendido que este es el clasificador que se nos pedia.

## Creacion del clasificador e indica su error

In [15]:
#Creamos los datos con de entrenamiento y el test
x=data.drop(["radioterapia_yes"],axis=1)
y=data["radioterapia_yes"]
#80% para el training 20%para el test
x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.2,random_state=0)
x

grado_maligno  clase_20-29  clase_30-39  clase_40-49  \
no-recurrence-events              3            0            1            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            1   
no-recurrence-events              3            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              3            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              3            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            1            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              2            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              3            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            1            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              1            0            0            0   
no-recurrence-events              2            0            0            1   
no-recurrence-events              1            0            0            0   
no-recurrence-events              1            0            0            1   
no-recurrence-events              1            0            0            1   
no-recurrence-events              2         

In [16]:
#Creacion del modelo
bayes=GaussianNB()
bayes = bayes.fit(x,y)
#Error y precision del modelo
print('Precision: '+ str(bayes.score(x_test,y_test)))
print('Error: '+ str(1-bayes.score(x_test,y_test)))

Precision: 0.4634146341463415
Error: 0.5365853658536586


In [17]:
bayes.class_prior_#Probabilidad de las clases

array([0.8159204, 0.1840796])

In [18]:
bayes.class_count_#Numero de samples por cada label /target

array([164.,  37.])

In [19]:
bayes.classes_#Labels/target

array([0, 1], dtype=uint8)

Podemos ver que hay 164 samples para el target 0 mientras que solo hay 37 samples para el target 1, esto quiere decir que los datos estan desbalanceados y esto puede explicar la poca precision del modelo dado que entrena con muchos mas datos del target 0 que del target 1.

## Clasificar 5 pacientes nuevos

In [20]:
test#Utilizamos los datos que hemos quitado antes del dataset.

grado_maligno  clase_20-29  clase_30-39  clase_40-49  \
recurrence-events              2            0            1            0   
recurrence-events              3            0            1            0   
recurrence-events              1            0            0            0   
recurrence-events              3            0            0            1   
recurrence-events              3            0            0            0   

                   clase_50-59  clase_60-69  clase_70-79  edad_ge40  \
recurrence-events            0            0            0          0   
recurrence-events            0            0            0          0   
recurrence-events            0            1            0          1   
recurrence-events            0            0            0          1   
recurrence-events            1            0            0          1   

                   edad_lt40  edad_premeno  ...  capsulas_no  capsulas_yes  \
recurrence-events          0             1  ...            1             0   
recurrence-events          0             1  ...            1             0   
recurrence-events          0             0  ...            1             0   
recurrence-events          0             0  ...            1             0   
recurrence-events          0             0  ...            1             0   

                   pecho_left  pecho_right  zona_pecho_central  \
recurrence-events           1            0                   0   
recurrence-events           1            0                   0   
recurrence-events           0            1                   0   
recurrence-events           1            0                   0   
recurrence-events           1            0                   0   

                   zona_pecho_left_low  zona_pecho_left_up  \
recurrence-events                    0                   1   
recurrence-events                    0                   1   
recurrence-events                    0                   1   
recurrence-events                    1                   0   
recurrence-events                    1                   0   

                   zona_pecho_right_low  zona_pecho_right_up  radioterapia_yes  
recurrence-events                     0                    0                 0  
recurrence-events                     0                    0                 1  
recurrence-events                     0                    0                 0  
recurrence-events                     0                    0                 0  
recurrence-events                     0                    0                 0  

[5 rows x 38 columns]

In [21]:
#Test clasificar 5 rows
x_test=test.drop(["radioterapia_yes"],axis=1)
y_test=test["radioterapia_yes"]

In [22]:
#Pasamos los datos nuevos por el clasificador y mostramos la matriz de confusio
confusion_matrix(bayes.predict(x_test), y_test)

array([[0, 0],
       [4, 1]])

Podemos ver que clasifica incorrectamente todos los casos positivos, sin embargo, clasifica correctamente los casos negativos.
Esto tiene sentido dado que ha entrenado con muchos mas datos con el target 0 que con el target 1.

# Referencias
https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.GaussianNB.html
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.crosstab.html